inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -0,0 +1,370 @@
+    int id = 0;
+    for (auto node : f.get_ordered_ops()) {
+        layer_ids[node.get()] = id++;
+    }
+    return layer_ids;
+}
+
+const std::vector<Edge> create_edge_mapping(
+    const std::map<ngraph::Node*, int>& layer_ids, const ngraph::Function& f) {
+    std::vector<Edge> edges;
+    for (auto node : f.get_ordered_ops()) {
+        if (dynamic_cast<ngraph::op::Parameter*>(node.get()) != nullptr) {
+            continue;
+        }
+
+        for (auto i : node->inputs()) {
+            auto source_output = i.get_source_output();
+            auto source_node = source_output.get_node();
+            auto current_node = i.get_node();
+
+            NGRAPH_CHECK(layer_ids.find(source_node) != layer_ids.end(),
+                         "Internal error.");
+            NGRAPH_CHECK(layer_ids.find(current_node) != layer_ids.end(),
+                         "Internal error.");
+
+            Edge e{};
+            e.from_layer = layer_ids.find(source_node)->second;
+            e.from_port =
+                source_node->get_input_size() + source_output.get_index();
+            e.to_layer = layer_ids.find(current_node)->second;
+            e.to_port = i.get_index();
+            edges.push_back(e);
+        }
+    }
+    std::sort(begin(edges), end(edges),
+              [](const Edge& a, const Edge& b) -> bool {
+                  return a.from_layer < b.from_layer;
+              });
+    return edges;
+}
+
+// TODO: refactor to Vistor API when Constant will be supporting it
+ConstantAtributes dump_constant_data(std::vector<uint8_t>& bin,
+                                     const ngraph::op::Constant& c) {
+    NGRAPH_CHECK(c.get_output_partial_shape(0.).is_static(),
+                 "Dynamic shape of ", c.get_name(), " (", c.get_type_name(),
+                 ") ", "is not supported.");
+
+    ConstantAtributes attr;
+    const uint8_t* p = reinterpret_cast<const uint8_t*>(c.get_data_ptr());
+    attr.size = ngraph::shape_size(c.get_shape()) * c.get_element_type().size();
+    attr.offset = bin.size();
+    bin.insert(end(bin), p, p + attr.size);
+    return attr;
+}
+
+std::string get_opset_name(const ngraph::Node* n) {
+    auto opsets = std::array<std::reference_wrapper<const ngraph::OpSet>, 5>{
+        ngraph::get_opset1(), ngraph::get_opset2(), ngraph::get_opset3(),
+        ngraph::get_opset4(), ngraph::get_opset5()};
+
+    // return the oldest opset name where node type is present
+    for (int idx = 0; idx < opsets.size(); idx++) {
+        int number = idx + 1;
+        if (opsets[idx].get().contains_op_type(n)) {
+            return "opset" + std::to_string(number);
+        }
+    }
+    return "experimental";
+}
+
+std::string get_type_name(const ngraph::Node* n) {
+    std::string name = n->get_type_name();
+    NGRAPH_CHECK(name != "GenericIE", "Type of ", n->get_name(), " (",
+                 n->get_type_name(), ") ", "is not supported.");
+
+    const std::map<std::string, std::string> translator = {
```

Comment:
To all reviewers. Here I am translating operation type names from ngraph convention to IR convention. Example is Constant (ngraph name) and Const (IR name). If you happen to now that this names differs also for different types, please provide their names here. I'll add it to the code.

Reply:
Could you add a comment about it in the code?

Reply:
Done in this [commit](https://github.com/openvinotoolkit/openvino/pull/2579/commits/a5a2e9e52b7a57b922003a516e53b5091f97383f).